Dockerfile

```diff
@@ -62,6 +62,20 @@ COPY config config
 COPY devtools devtools
 RUN npm install
 
+# Fetch external Bazel artifacts.
+# Copy over the WORKSPACE file, everything it imports, and bazelisk.
+COPY tools/bazelisk* tools/
+COPY build_defs/emscripten build_defs/emscripten
+COPY build_defs/kotlin_native build_defs/kotlin_native
+COPY emscripten_cache emscripten_cache
+COPY .bazelignore \
+     .bazelversion \
+     WORKSPACE \
+     BUILD.bazel \
+     maven_install.json \
+     ./
+RUN ./tools/bazelisk fetch @maven//...
```

Comment:
Isn't the Dockerfile processed by GCB every time, so what if this RUN command fails to fetch the artifacts?

Reply:
This will only rerun if any of the previous steps changed. Otherwise, we will get an image for this layer from the google container registry (gcr) cache.

Reply:
By "previous steps changed" I assume the COPY command detecting a new maven_install.json would count as a change? Ok, this sounds like a very good change.

Reply:
Yep, we will only try to download if `maven_install.json` changed.